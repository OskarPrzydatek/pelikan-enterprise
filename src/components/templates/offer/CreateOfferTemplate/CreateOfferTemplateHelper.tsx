import { useForm } from 'react-hook-form';

import { CreateOfferTemplate } from '~/components/templates';
import { IOffer } from '~/models';
import { StyledComponentsProvider } from '~/providers';

interface ICreateOfferTemplateHelper {
  onSubmitCreateOffer: () => void;
}

export const CreateOfferTemplateHelper: React.FC<
  ICreateOfferTemplateHelper
> = ({ onSubmitCreateOffer }: ICreateOfferTemplateHelper) => {
  const methods = useForm<IOffer>();

  return (
    <StyledComponentsProvider>
      <CreateOfferTemplate
        createOfferFormMethods={methods}
        hotelOptions={[
          { value: 'HOTEL_A', label: 'Hotel A' },
          { value: 'HOTEL_B', label: 'Hotel B' },
          { value: 'HOTEL_C', label: 'Hotel C' },
        ]}
        transportOptions={[
          { value: 'TRANSPORT_A', label: 'Transport A' },
          { value: 'TRANSPORT_B', label: 'Transport B' },
        ]}
        onSubmitCreateOffer={onSubmitCreateOffer}
      />
    </StyledComponentsProvider>
  );
};
