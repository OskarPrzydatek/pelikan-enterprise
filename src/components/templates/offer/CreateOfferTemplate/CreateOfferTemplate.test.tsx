import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { CreateOfferTemplateHelper } from './CreateOfferTemplateHelper';

const mockOnSubmitCreateOffer = vi.fn();

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual<object>('react-router-dom');

  return {
    ...actual,
    useNavigate: vi.fn(),
  };
});

describe('CreateOfferTemplate', () => {
  test('component snapshot', () => {
    const view = render(
      <CreateOfferTemplateHelper
        onSubmitCreateOffer={mockOnSubmitCreateOffer}
      />
    );
    expect(view).toMatchSnapshot();
  });
});
