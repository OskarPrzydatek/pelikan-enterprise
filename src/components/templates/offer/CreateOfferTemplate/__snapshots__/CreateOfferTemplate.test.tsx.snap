// Vitest Snapshot v1

exports[`CreateOfferTemplate > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iAEawV eSwvqh"
      >
        <form
          class="sc-gikAfH cxczgd"
        >
          <h2
            class="sc-dkrFOg kaKqDq"
          >
            Dodaj ofertę
          </h2>
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <input
                class="sc-dIfARi cqofnP"
                data-testid="create-offer-name"
                id="name"
                name="name"
                placeholder="Nazwa oferty"
              />
            </div>
          </div>
          <div
            class="sc-gKPRtg gOHhkn"
          >
            <div
              class="sc-bjfHbI kTUcsA"
            >
              <div
                class="sc-idXgbr enkruV"
              >
                <input
                  class="sc-dIfARi cqofnP"
                  data-testid="create-offer-number-of-tickets"
                  id="numberOfGuests"
                  name="numberOfGuests"
                  placeholder="Ilość gości"
                />
              </div>
            </div>
            <div
              class="sc-dmctIk iThEeJ"
            >
              <div
                class="sc-kgTSHT fmvgAY"
              >
                <span
                  class="sc-bBABsx kBxwHb"
                >
                  <span
                    class="sc-eDvSVe ktaJny"
                  >
                    <svg
                      data-testid="chevron-icon"
                      fill="none"
                      viewBox="0 0 27 16"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2 2L13.5 13L25 2"
                        stroke="#013059"
                        stroke-width="3"
                      />
                    </svg>
                  </span>
                </span>
                <select
                  class="sc-iveFHk eqISZE"
                  data-testid="create-offer-transport"
                  name="transport"
                >
                  <option
                    class="sc-cwSeag"
                    disabled=""
                    selected=""
                    value=""
                  >
                    Transport
                  </option>
                  <option
                    class="sc-cwSeag"
                    value="TRANSPORT_A"
                  >
                    Transport A
                  </option>
                  <option
                    class="sc-cwSeag"
                    value="TRANSPORT_B"
                  >
                    Transport B
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="sc-gKPRtg gOHhkn"
          >
            <div
              class="sc-bjfHbI kTUcsA"
            >
              <div
                class="sc-idXgbr enkruV"
              >
                <input
                  class="sc-dIfARi cqofnP"
                  data-testid="create-offer-destination"
                  id="destination"
                  name="destination"
                  placeholder="Miejsce docelowe"
                />
              </div>
            </div>
            <div
              class="sc-dmctIk iThEeJ"
            >
              <div
                class="sc-kgTSHT fmvgAY"
              >
                <span
                  class="sc-bBABsx kBxwHb"
                >
                  <span
                    class="sc-eDvSVe ktaJny"
                  >
                    <svg
                      data-testid="chevron-icon"
                      fill="none"
                      viewBox="0 0 27 16"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2 2L13.5 13L25 2"
                        stroke="#013059"
                        stroke-width="3"
                      />
                    </svg>
                  </span>
                </span>
                <select
                  class="sc-iveFHk eqISZE"
                  data-testid="create-offer-hotel"
                  name="hotel"
                >
                  <option
                    class="sc-cwSeag"
                    disabled=""
                    selected=""
                    value=""
                  >
                    Hotel
                  </option>
                  <option
                    class="sc-cwSeag"
                    value="HOTEL_A"
                  >
                    Hotel A
                  </option>
                  <option
                    class="sc-cwSeag"
                    value="HOTEL_B"
                  >
                    Hotel B
                  </option>
                  <option
                    class="sc-cwSeag"
                    value="HOTEL_C"
                  >
                    Hotel C
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="sc-gKPRtg gOHhkn"
          >
            <div
              class="sc-ftTHYK gCnWqt"
            >
              <div
                class="sc-pyfCe dPDCSN"
              >
                <input
                  class="sc-jrcTuL iGTWLz"
                  data-testid="create-offer-form"
                  min="2023-08-02"
                  name="termFrom"
                  type="date"
                />
              </div>
            </div>
            <div
              class="sc-ftTHYK gCnWqt"
            >
              <div
                class="sc-pyfCe dPDCSN"
              >
                <input
                  class="sc-jrcTuL iGTWLz"
                  data-testid="create-offer-to"
                  min="2023-08-02"
                  name="termTo"
                  type="date"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-ksBlkl gBSrBs"
          >
            <div
              class="sc-hBxehG eptxaL"
            >
              <textarea
                class="sc-fEXmlR kcJGhn"
                data-testid="create-offer-desription"
                id="description"
                name="description"
                placeholder="Opis"
              />
            </div>
          </div>
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <input
                class="sc-dIfARi cqofnP"
                data-testid="create-offer-price"
                id="price"
                name="price"
                placeholder="Cena"
              />
            </div>
          </div>
          <input
            class="sc-ezOQGI cvtWIm"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iAEawV eSwvqh"
    >
      <form
        class="sc-gikAfH cxczgd"
      >
        <h2
          class="sc-dkrFOg kaKqDq"
        >
          Dodaj ofertę
        </h2>
        <div
          class="sc-bjfHbI kTUcsA"
        >
          <div
            class="sc-idXgbr enkruV"
          >
            <input
              class="sc-dIfARi cqofnP"
              data-testid="create-offer-name"
              id="name"
              name="name"
              placeholder="Nazwa oferty"
            />
          </div>
        </div>
        <div
          class="sc-gKPRtg gOHhkn"
        >
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <input
                class="sc-dIfARi cqofnP"
                data-testid="create-offer-number-of-tickets"
                id="numberOfGuests"
                name="numberOfGuests"
                placeholder="Ilość gości"
              />
            </div>
          </div>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <span
                class="sc-bBABsx kBxwHb"
              >
                <span
                  class="sc-eDvSVe ktaJny"
                >
                  <svg
                    data-testid="chevron-icon"
                    fill="none"
                    viewBox="0 0 27 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2L13.5 13L25 2"
                      stroke="#013059"
                      stroke-width="3"
                    />
                  </svg>
                </span>
              </span>
              <select
                class="sc-iveFHk eqISZE"
                data-testid="create-offer-transport"
                name="transport"
              >
                <option
                  class="sc-cwSeag"
                  disabled=""
                  selected=""
                  value=""
                >
                  Transport
                </option>
                <option
                  class="sc-cwSeag"
                  value="TRANSPORT_A"
                >
                  Transport A
                </option>
                <option
                  class="sc-cwSeag"
                  value="TRANSPORT_B"
                >
                  Transport B
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="sc-gKPRtg gOHhkn"
        >
          <div
            class="sc-bjfHbI kTUcsA"
          >
            <div
              class="sc-idXgbr enkruV"
            >
              <input
                class="sc-dIfARi cqofnP"
                data-testid="create-offer-destination"
                id="destination"
                name="destination"
                placeholder="Miejsce docelowe"
              />
            </div>
          </div>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <span
                class="sc-bBABsx kBxwHb"
              >
                <span
                  class="sc-eDvSVe ktaJny"
                >
                  <svg
                    data-testid="chevron-icon"
                    fill="none"
                    viewBox="0 0 27 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2L13.5 13L25 2"
                      stroke="#013059"
                      stroke-width="3"
                    />
                  </svg>
                </span>
              </span>
              <select
                class="sc-iveFHk eqISZE"
                data-testid="create-offer-hotel"
                name="hotel"
              >
                <option
                  class="sc-cwSeag"
                  disabled=""
                  selected=""
                  value=""
                >
                  Hotel
                </option>
                <option
                  class="sc-cwSeag"
                  value="HOTEL_A"
                >
                  Hotel A
                </option>
                <option
                  class="sc-cwSeag"
                  value="HOTEL_B"
                >
                  Hotel B
                </option>
                <option
                  class="sc-cwSeag"
                  value="HOTEL_C"
                >
                  Hotel C
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="sc-gKPRtg gOHhkn"
        >
          <div
            class="sc-ftTHYK gCnWqt"
          >
            <div
              class="sc-pyfCe dPDCSN"
            >
              <input
                class="sc-jrcTuL iGTWLz"
                data-testid="create-offer-form"
                min="2023-08-02"
                name="termFrom"
                type="date"
              />
            </div>
          </div>
          <div
            class="sc-ftTHYK gCnWqt"
          >
            <div
              class="sc-pyfCe dPDCSN"
            >
              <input
                class="sc-jrcTuL iGTWLz"
                data-testid="create-offer-to"
                min="2023-08-02"
                name="termTo"
                type="date"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-ksBlkl gBSrBs"
        >
          <div
            class="sc-hBxehG eptxaL"
          >
            <textarea
              class="sc-fEXmlR kcJGhn"
              data-testid="create-offer-desription"
              id="description"
              name="description"
              placeholder="Opis"
            />
          </div>
        </div>
        <div
          class="sc-bjfHbI kTUcsA"
        >
          <div
            class="sc-idXgbr enkruV"
          >
            <input
              class="sc-dIfARi cqofnP"
              data-testid="create-offer-price"
              id="price"
              name="price"
              placeholder="Cena"
            />
          </div>
        </div>
        <input
          class="sc-ezOQGI cvtWIm"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
